.build/run-tests.sh

```diff
@@ -84,7 +87,9 @@ _build_all_dtest_jars() {
     fi
 
     if [ -d ${TMP_DIR}/cassandra-dtest-jars ] && [ "https://github.com/apache/cassandra.git" == "$(git -C ${TMP_DIR}/cassandra-dtest-jars remote get-url origin)" ] ; then
-        until git -C ${TMP_DIR}/cassandra-dtest-jars fetch --quiet origin ; do echo "git pull failed… trying again… " ; done
+      if [ "x" == "x${OFFLINE}" ] ; then
+        until git -C ${TMP_DIR}/cassandra-dtest-jars fetch --quiet origin ; do echo "git fetch failed… trying again… " ; done
```

Comment:
Can this infinite loop / hang? And if so maybe we constrain it and explain why the need?

Reply:
it can, but you get the echo statements so it's visible to the user. they can ctl-c it.

this poormans approach is repeated in a number of places.  improving it (which is smart) should be done together.  and maybe just rewriting bash altogether into something non-shit